### PR TITLE
bump: minimum go version set to 1.20

### PR DESCRIPTION
**Pull request by IceCodeNew**
Since the caddy@master has already requested a minimum go version of 1.20, I would like to align with the latest updates from the upstream.
Doing so may cause caddy-l4 to have problems when compiling it against the current stable version of caddy. I created a `stable` branch for that case.
If it makes sense to you, please consider making the `stable` branch the default branch of this project.

Please also take a look at https://github.com/mholt/caddy-l4/commit/9627d11de27bb96c34c30402b18efed1043e51da

### CODE REVIEW

**Review by mholt — APPROVED** [review]
Thanks, we actually are moving to Go 1.20 in Caddy with 2.7 (it's already on 1.20 on master).

Is it OK if I merge this then? (I don't think we'll need the `stable` branch.)

File: .github/dependabot.yml [hunk 1]
```diff
@@ -10,9 +10,11 @@ updates:
     directory: "/"
     schedule:
       interval: "weekly"
+    target-branch: "master"
```
[the remark this answers is not in the document]
Comment (IceCodeNew) [reply .github/dependabot.yml:13]: > What do these lines do differently?


https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#target-branch
> By default, Dependabot checks for manifest files on the default branch and raises pull requests for version updates against this branch. Use target-branch to specify a different branch for manifest files and for pull requests. When you use this option, the settings for this package manager will no longer affect any pull requests raised for security updates.

I intended to make sure that both the `stable` branch and the `master` branch will receive update notifications.

If we get to the consensus that the `stable` branch is needless, it is fine to not set the `target-branch` option.

**Comment by IceCodeNew** [comment]
请帮我美化下面的英语句子：

I delved further into the previous error when attempting to compile caddy-l4 against the latest stable release of Caddy. Turns out this problem could be resolved by setting the go environment to version 1.20.
It now looks like this project is good to go without a `stable` branch. I will delete that branch and modify this pull request to be ready for merging.

**Review by mohammed90 — APPROVED** [review]

**Comment by IceCodeNew** [comment]
@mholt @mohammed90 You may merge this PR at your convenient.

File: .github/dependabot.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mholt) [reply .github/dependabot.yml]: Gotcha. I think we don't plan on keeping a "stable" branch at this time, so we can probably remove this for simplicity.

[the remark this answers is not in the document]
Comment (mholt) [reply .github/dependabot.yml]: Oh, nevermind -- already done... missed that...

**Comment by mholt** [comment]
Thank you!

**Pull request merged**